conda_forge_tick/update_recipe/version.py

```diff
@@ -625,9 +676,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
     return data["updated"], data["errors"]
 
 
-def update_version(raw_meta_yaml, version, hash_type="sha256"):
+def update_version_v1(
+    feedstock_dir: str, version: str, hash_type: str
+) -> (str | None, set[str]):
     """Update the version in a recipe.
 
+    Parameters
+    ----------
+    feedstock_dir : str
+        The feedstock directory to update.
+    version : str
+        The new version of the recipe.
+    hash_type : str
+        The kind of hash used on the source.
+
+    Returns
+    -------
+    updated_meta_yaml : str or None
+        The updated meta.yaml. Will be None if there is an error.
+    errors : set of str
+    """
+    # extract all the URL sources from a given recipe / feedstock directory
+    from rattler_build_conda_compat.loader import load_yaml
+    from rattler_build_conda_compat.recipe_sources import render_all_sources
+
+    feedstock_dir = Path(feedstock_dir)
+    recipe_path = feedstock_dir / "recipe" / "recipe.yaml"
+    recipe_yaml = load_yaml(recipe_path.read_text())
```

Comment:
Yet other variables being used here
```suggestion
    recipe_path = feedstock_dir / "recipe" / "recipe.yaml"
    recipe_text = recipe_path.read_text()
    recipe_yaml = load_yaml(recipe_text)
```

Reply:
might also want to rename `recipe_yaml` to `recipe_dict` or `recipe_info`, because after `load_yaml`, it's not actually a YAML in any reasonable sense of the word anymore.

Reply:
A (ruamel) YAML is also not exactly a dict so might be better to keep it as is.